### PR TITLE
scripts: board_v1_to_v2: License&copyright + SOC_SERIES fix

### PR DESCRIPTION
**Pull request by erwango**
This PR is adding 2 changes:

- When creating file Kconfig.<board> a copyright is required as otherwise an issue will be reported by CI scan code tool.
Always add License (SPDX) and add the option to put a copyright.

- Don't add select CONFIG_SOC_SERIES_FOO as part of board description. It is done at soc level.

### CODE REVIEW

File: scripts/utils/board_v1_to_v2.py [hunk 1]
```diff
@@ -157,11 +158,15 @@ def board_v1_to_v2(board_root, board, new_board, group, vendor, soc, variants):
 
     print(f"Creating or updating Kconfig.{new_board}...")
     new_board_kconfig_file = new_board_path / f"Kconfig.{new_board}"
-    selects = "\n\t" + "\n\t".join(["select " + setting for setting in board_soc_settings])
+    header = "# SPDX-License-Identifier: Apache-2.0\n"
+    if copyright is not None:
+        header = header + "# Copyright (c) 2024 " + copyright + "\n"
```
Comment (gmarull) [inline scripts/utils/board_v1_to_v2.py:163]: hardcoding year isn't likely a good idea (although... I hope we have finished by 2025)

File: scripts/utils/board_v1_to_v2.py [hunk 2]
```diff
@@ -157,11 +158,15 @@ def board_v1_to_v2(board_root, board, new_board, group, vendor, soc, variants):
 
     print(f"Creating or updating Kconfig.{new_board}...")
     new_board_kconfig_file = new_board_path / f"Kconfig.{new_board}"
-    selects = "\n\t" + "\n\t".join(["select " + setting for setting in board_soc_settings])
+    header = "# SPDX-License-Identifier: Apache-2.0\n"
```
Comment (gmarull) [inline scripts/utils/board_v1_to_v2.py:161]: I wonder if we should just re-use copyright from the deleted "Kconfig.board"

File: scripts/utils/board_v1_to_v2.py (hunk 1 above)
Comment (erwango) [reply scripts/utils/board_v1_to_v2.py:163]: Making it configurable sounded a bit over-engineered given the condition of usage.

File: scripts/utils/board_v1_to_v2.py (hunk 2 above)
Comment (erwango) [reply scripts/utils/board_v1_to_v2.py:161]: You're right. I'll have another shot.

Comment (erwango) [reply scripts/utils/board_v1_to_v2.py:161]: @gmarull What if we just:
- move and rename the existing Kconfig.board
- drop the "bool"
- s/depends on/select

Comment (gmarull) [reply scripts/utils/board_v1_to_v2.py:161]: some selects will need to be removed (e.g. all SOC_SERIES*), I need to update the script for that. So not sure if easier to update the file or regenerate it.

Comment (erwango) [reply scripts/utils/board_v1_to_v2.py:161]: On some boards, we'd need to remove the help as well indeed. See https://github.com/zephyrproject-rtos/zephyr/pull/67737/files

Comment (gmarull) [reply scripts/utils/board_v1_to_v2.py:161]: so then I'd just grab the copyright and create a new file as we do now

File: scripts/utils/board_v1_to_v2.py (hunk 1 above)
Comment (erwango) [reply scripts/utils/board_v1_to_v2.py]: Resolved as I'm now retrieving previous Copyright.

File: scripts/utils/board_v1_to_v2.py [hunk 3]
```diff
@@ -156,19 +156,27 @@ def board_v1_to_v2(board_root, board, new_board, group, vendor, soc, variants):
...
+    board_kconfig_file = new_board_path / "Kconfig.board"
+    with open(board_kconfig_file) as f:
+        for line in f.readlines():
+            if "Copyright" in line:
+                copyright = line
     new_board_kconfig_file = new_board_path / f"Kconfig.{new_board}"
-    selects = "\n\t" + "\n\t".join(["select " + setting for setting in board_soc_settings])
+    header = "# SPDX-License-Identifier: Apache-2.0\n"
+    if copyright is not None:
+        header = header + copyright
```
Comment (gmarull) [inline scripts/utils/board_v1_to_v2.py:167]: shouldn't this be 
```suggestion
    if copyright is not None:
        header = copyright + header
```
?

Comment (erwango) [reply scripts/utils/board_v1_to_v2.py:167]: Right.

**Review by gmarull — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

File: scripts/utils/board_v1_to_v2.py (hunk 2 above)
Comment (erwango) [reply scripts/utils/board_v1_to_v2.py:165]: > some selects will need to be removed (e.g. all SOC_SERIES*), I need to update the script for that. So not sure if easier to update the file or regenerate it.

Do you think about something like follows ?
```
-            m = re.match(r"^CONFIG_(SOC_[A-Z0-9_]+).*$", line)
+            m = re.match(r"^CONFIG_(SOC_(?!SERIES_)[A-Z0-9_]+).*$", line)
```

Comment (gmarull) [reply scripts/utils/board_v1_to_v2.py:165]: yes, willing to add a patch :-)?

Comment (erwango) [reply scripts/utils/board_v1_to_v2.py:165]: Done ;)

**Review by gmarull — APPROVED** [review]

**Comment by erwango** [comment]
Set as DNM as proposed change is not sufficient to drop `CONFIG_SOC_SERIES_STM32F0X` line in board_defconfig file

**Comment by erwango** [comment]
@gmarull Hope it is fine now.

**Review by gmarull — APPROVED** [review]

**Pull request merged**